libethashseal/Ethash.cpp

```diff
@@ -157,6 +157,8 @@ void Ethash::verifyTransaction(ImportRequirements::value _ir, TransactionBase co
 		}
 		else
 			_t.checkChainId(-4);
+		if (_bi.number() < chainParams().u256Param("metropolisForkBlock") && _t.hasZeroSignature())
```

Comment:
We should better leave this check only in `SealEngineFace::verifyTransaction` (base class) and call `SealEngineFace::verifyTransaction(_ir, _t, _bi)` in the beginning of `Ethash::verifyTransaction`